src/app/routes/applications/new-role/new-role.component.spec.ts

```diff
@@ -1,25 +1,259 @@
+import { FormBuilder, ReactiveFormsModule, Validators } from '@angular/forms';
 
-import { NewRoleComponent } from './new-role.component';
+import { provideMockStore } from '@ngrx/store/testing';
 
-xdescribe('NewRoleComponent', () => {
+import { NewRoleComponent, RoleType } from './new-role.component';
+import { IamService } from '../../../shared/services/iam.service';
+import { ToastrService } from 'ngx-toastr';
+import { MAT_DIALOG_DATA, MatDialog, MatDialogRef } from '@angular/material/dialog';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { PreconditionTypes } from 'iam-client-lib';
+
+describe('NewRoleComponent', () => {
   let component: NewRoleComponent;
   let fixture: ComponentFixture<NewRoleComponent>;
+  const fb = new FormBuilder();
+  const matDialogSpy = jasmine.createSpyObj('MatDialog',
+      [
+        'closeAll',
+      ]);
+  const toastrSpy = jasmine.createSpyObj('ToastrService',
+      [
+        'success',
+        'error'
+      ]);
+  const iamSpy = jasmine.createSpyObj('iam', [
+    'getDid',
+    'checkExistenceOfDomain',
+    'isOwner',
+    'getRoleDIDs',
+    'createRole',
+    'setRoleDefinition',
+    'getENSTypesBySearchPhrase'
+  ]);
+  const iamServiceSpy = jasmine.createSpyObj('iam', [
+    'isAlphaNumericOnly',
+  ]);
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
-      declarations: [ NewRoleComponent ]
+      declarations: [ NewRoleComponent ],
+      imports: [ReactiveFormsModule],
+      providers: [
+        provideMockStore(),
+        { provide: IamService, useValue: { ...iamServiceSpy, iam: iamSpy}},
+        { provide: ToastrService, useValue: toastrSpy },
+        { provide: MatDialogRef, useValue: {} },
+        { provide: MatDialog, useValue: matDialogSpy },
+        { provide: MAT_DIALOG_DATA, useValue: {} }
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
     })
     .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(NewRoleComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
+    // fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  describe('should be run', () => {
+
+    it('removeDid', () => {
+      const i = 1;
+      component.issuerList = ['1', '2', '3'];
+      component.removeDid(i);
+
+      expect(component.issuerList).toEqual(['1', '3']);
+
+      component.issuerList = ['1'];
+      component.removeDid(0);
+
+      expect(component.issuerList).toEqual(['1']);
+    });
+
+    it('formResetHandler', () => {
+      spyOn(component.fieldsForm, 'reset');
+
+      component.formResetHandler();
+
+      expect(component.fieldsForm.reset).toHaveBeenCalled();
+    });
+
+    it('dataSourceChangeHandler', () => {
+      const data = ['test value'];
+
+      component.dataSourceChangeHandler(data);
+
+      expect(component.dataSource.data).toEqual(data);
+    });
+
+    it('ngAfterViewInit', () => {
```

Comment:
This test just checking if method have been called but do not test behavior of the component. Imo you can delete it as well.